### ISSUE TITLE
PuTTY client

### ISSUE THREAD

**Issue by HarelAshwal** [kstafford3/ssh2-shell-server#1]
LK"I

Hi There,
Tried to use the passowrd example , but no luck with PuTTY.

Error Report:

Listening on port 5151...
Error! { err: { Error: No supported authentication methods available
    at onDISCONNECT (C:\projects_node\ssh_test\ssh_test\ssh_test\node_modules\ssh2-streams\lib\ssh.js:2202:15)
    at SSH2Stream.<anonymous> (C:\projects_node\ssh_test\ssh_test\ssh_test\node_modules\ssh2-streams\lib\ssh.js:203:5)
    at emitMany (events.js:146:13)
    at SSH2Stream.emit (events.js:223:7)
    at parsePacket (C:\projects_node\ssh_test\ssh_test\ssh_test\node_modules\ssh2-streams\lib\ssh.js:3748:10)
    at SSH2Stream._transform (C:\projects_node\ssh_test\ssh_test\ssh_test\node_modules\ssh2-streams\lib\ssh.js:669:13)
    at SSH2Stream.Transform._read (_stream_transform.js:186:10)
    at SSH2Stream._read (C:\projects_node\ssh_test\ssh_test\ssh_test\node_modules\ssh2-streams\lib\ssh.js:251:15)
    at SSH2Stream.Transform._write (_stream_transform.js:174:12)
    at doWrite (_stream_writable.js:387:12) code: 14 },
  client:
   Client {
     domain: null,
     _events:
      { ready: [Array],
        authentication: [Function],
        end: [Function],
        error: [Function] },
     _eventsCount: 4,
     _maxListeners: undefined,
     _sshstream:
      SSH2Stream {
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _eventsCount: 15,
        _maxListeners: undefined,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: true,
        _transformState: [Object],
        _needContinue: false,
        bytesReceived: 256,
        bytesSent: 112,
        debug: [Function: DEBUG_NOOP],
        server: true,
        maxPacketSize: 35000,
        remoteBugs: 1,
        config: [Object],
        _state: [Object] },
     _channels: {},
     _curChan: -1,
     _sock:
      Socket {
        connecting: false,
        _hadError: false,
        _handle: [Object],
        _parent: null,
        _host: null,
        _readableState: [Object],
        readable: true,
        domain: null,
        _events: [Object],
        _eventsCount: 7,
        _maxListeners: undefined,
        _writableState: [Object],
        writable: true,
        allowHalfOpen: false,
        _bytesDispatched: 1281,
        _sockname: null,
        _pendingData: null,
        _pendingEncoding: '',
        server: [Object],
        _server: [Object],
        read: [Function],
        _consuming: true,
        _peername: [Object],
        [Symbol(asyncId)]: 8,
        [Symbol(bytesRead)]: 0 },
     noMoreSessions: false,
     authenticated: false },
  session: null }
Session Ended

Code:

```

const { ShellServer, Authenticators } = require('../lib/index');

const fs = require('fs');
const readline = require('readline');

const keyFile = fs.readFileSync('./test_keys/test_key');
const PORT = 5151;

const QUIT_KEYWORDS = [ 'quit', 'exit', 'q' ];

const server = new ShellServer({
  hostKeys: [ keyFile ],
  port: PORT,
},
Authenticators.authenticateByPassword(checkPassword));

// Authenticators.authenticateByPassword(checkPassword)); // Alternative authentication

server.on('session-created', (session) => {
  console.log('Session Created');
  session.on('stream-initialized', (stream) => {
    console.log('Stream Initialized');
    stream.write('Welcome to the password-protected server!\r\n');
    const rl = readline.createInterface({
      input: stream,
      output: stream,
      prompt: '>',
    });
    rl.prompt();
    rl.on('line', (line) => {
      console.log(`${session.username}: ${line}`);
      stream.write(`${line}\r\n`);
      if (QUIT_KEYWORDS.indexOf(line.toLowerCase()) > -1) {
        rl.close();
      }
      rl.prompt();
    });

    rl.on('close', () => {
      stream.write('Ok, well, I guess I\'ll talk to you later, then.\r\n');
      stream.close();
    });
  });
});

server.on('session-ended', (e) => {
  console.log('Session Ended');
});

server.on('error', (e) => {
  console.error('Error!', e);
});

server.listen().then(() => {
  console.log(`Listening on port ${PORT}...`);
});

function checkPassword(username, password) {
  return password === 'password'; // secure af
}





```

**Comment by kstafford3** [comment]
Hi HarelAshwal,

Thanks for filing this issue, it looks like the problem arises from the authenticators not returning a list of accepted authentication methods.

**Comment by kstafford3** [comment]
Resolved in v2.0.0
